CRM/Dedupe/Merger.php

```diff
@@ -841,33 +839,26 @@ public static function skipMerge($mainId, $otherId, &$migrationInfo, $mode = 'sa
         $fieldName = $locField[2];
         $fieldCount = $locField[3];
 
-        // Rule: resolve address conflict if any
+        // Rule: Catch address conflicts (same address type on both contacts)
         if ($fieldName == 'address') {
           $mainNewLocTypeId = $migrationInfo['location_blocks'][$fieldName][$fieldCount]['locTypeId'];
-          if (!empty($migrationInfo['main_loc_block']) &&
-              array_key_exists("main_address_{$mainNewLocTypeId}", $migrationInfo['main_loc_block'])) {
-            // main loc already has some address for the loc-type. Its a overwrite situation.
-            // look for next available loc-type
-            $newTypeId = NULL;
-            foreach ($allLocationTypes as $typeId => $typeLabel) {
-              if (!array_key_exists("main_address_{$typeId}", $migrationInfo['main_loc_block'])) {
-                $newTypeId = $typeId;
+          if (
+            isset($migrationInfo['main_details']['location_blocks']['address']) &&
```

Comment:
So, my understanding is that the array $migrationInfo['main_loc_block'] is the same as 
$migrationInfo['main_details']['location_blocks']['address'] - I'm kind of wishing I could QA the changes to the array without having this behaviour changing patch mixed in - mostly because I still think the behaviour is not quite there - per comments elsewhere and I think we have to have some unit tests before we can merge this